make/mt-config/mt-config

```diff
@@ -56,7 +56,12 @@ ifneq (,$(findstring debug,$(TARGET)))
 	XI_DEBUG_EXTRA_INFO ?= 1
 endif
 
-XI_COMPILER_FLAGS += -Wall -Werror
+# Settings that will work only on linux and only against clang-4.0 and greater
+ifneq (,$(findstring fuzz_test,$(CONFIG)))
+    XI_CONFIG_FLAGS += -fsanitize=address -fomit-frame-pointer -fsanitize-coverage=trace-pc-guard -g
+endif
+
+XI_COMPILER_FLAGS += -Wall
```

Comment:
But I didn't enable it, it was already enabled I've just moved the whole section. But in terms of -Wall I would leave it as it increases the quality of compiler tests - and this is the first line of defence. I would like to enable -pedantic but maybe some other time :)